__tests__/commands/install/integration-hoisting.js

```diff
@@ -52,3 +55,49 @@ test.concurrent(
     });
   },
 );
+
+test.concurrent('install hoister should remove newly hoisted dependencies from non-hoisted locations', (): Promise<
+  void,
+> => {
+  // Arrange (fixture):
+  //   /@s/x@1.0.0
+  //   /@s/y@1.0.0
+  //   /a@1.0.0
+  //   /b@1.0.0
+  //       /@s/x@1.5.0
+  //       /@s/y@2.0.0
+  // Act: Update dep versions using add
+  // Assert: newly hoisted dependencies are deleted from node_modules sub-trees
+  return runInstall({}, 'should-remove-newly-hoisted-@deps', async (config, reporter) => {
+    // assert initial setup is as expected
+    expect(await getPackageVersion(config, '@s/x')).toEqual('1.0.0');
+    expect(await getPackageVersion(config, '@s/y')).toEqual('1.0.0');
+    expect(await getPackageVersion(config, 'a')).toEqual('1.0.0');
+    expect(await getPackageVersion(config, 'b')).toEqual('1.0.0');
+    expect(await getPackageVersion(config, 'b/@s/x')).toEqual('1.5.0');
+    expect(await getPackageVersion(config, 'b/@s/y')).toEqual('2.0.0');
+
+    await add(config, reporter, {}, ['file:a/v2.0.0', 'file:b/v2.0.0']);
+
+    const getAllFilePaths = async currentPath => {
+      const lstat = await fs.lstat(currentPath);
+      if (lstat.isSymbolicLink() || lstat.isFile()) {
+        return [currentPath];
+      }
+
+      const files = await fs.readdir(currentPath);
+      const subFiles = files.map(file => getAllFilePaths(path.join(currentPath, file)));
+      return (await Promise.all(subFiles)).reduce((previous, current) => previous.concat(...current), []);
+    };
+    const allFilePathsInTestDirectory = await getAllFilePaths(config.cwd);
+    config.reporter.info(allFilePathsInTestDirectory.join('\n'));
```

Comment:
This reporter info call was for debugging the CI only, will remove it once the CI is passing